pkgs/build-support/rust/cargo-local-registry.nix

```diff
@@ -0,0 +1,33 @@
+{ fetchurl, stdenv }:
+
+let
+  version = "0.1.4";
+  platform =
+    if stdenv.system == "x86_64-linux"
```

Comment:
The reason I prefer this approach is it doesn't require any preprocessing or extra files. It simply reads the Cargo.toml file. I do think that those are fixable but this is also a nice quick improvement on the current state.